### DISCUSSION TITLE
SPO File versions

### DISCUSSION THREAD

**Discussion by Ranjith-2024** [pnp/powershell#4546] [Q&A]
Hello ,

We have multiple SPO sites impacted with huge amount of storage currently. Planned to delete the old file versions from each sites and to keep only the updated file version. Some of the sites are having 400 gb to 1200 GB of storage. (SPO Currently Ruuning out of storage space) We need to know the current file versions being used in each sites , so that it will hep us to know the existing file versions of each file / document per site and can decide which versions to be used and delete all other old versions. (We are working with MS)

We need a PnP Script to fetch above requirement / data/ report from each SPO Sites , Once we have the reports ready from each sites , decision will be taken accordingly for the deletions of Old versions. We may need a script at that point of time for deleting old versions from each sites as well.

Could you please have a look and help us with Scripts asap. OPened a case with PnP , no reply has been received yet. Really Appreciate your contributions..

**Comment by ToddKlindt** [comment]
The PnP Samples gallery is a great place to start looking for some help. There are a [bunch of script samples](https://pnp.github.io/script-samples/?query=version) that deal with discovering versions and deleting them. 

Good luck.

**Reply by Ranjith-2024** [reply]
Thanks Toddklindt ..will try
